include/clap/ext/draft/context-menu.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include "../../plugin.h"
+
+// This extension lets the host and plugin exchange menu items and let the plugin ask the host to
+// show its context menu.
+
+static CLAP_CONSTEXPR const char CLAP_EXT_CONTEXT_MENU[] = "clap.context-menu.draft/0";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// There can be different target kind for a context menu
+enum {
+   CLAP_CONTEXT_MENU_TARGET_KIND_GLOBAL,
+   CLAP_CONTEXT_MENU_TARGET_KIND_PARAM,
+};
+
+// Describes the context menu target
+typedef struct clap_context_menu_target
+{
+   uint32_t kind;
+   clap_id id;
+} clap_context_menu_target_t;
+
+// Flags for the context menu accelerator
+enum {
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_SHIFT = (1 << 0),
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_CONTROL = (1 << 1),
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_ALT = (1 << 2),
+
+   // Windows key or Command key
+   CLAP_CONTEXT_MENU_ACCELERATOR_MODIFIER_META1 = (1 << 3),
+};
+
+// Describes a single key strike with its modifiers
+typedef struct clap_context_menu_accelerator_key {
+   uint32_t flags;
+   uint32_t key; // ASCII key
+} clap_context_menu_accelerator_key_t;
+
+// Describes an accelerator which can consist of a key strike combo
+typedef struct clap_context_menu_accelerator {
+   uint32_t combo_size;
+   clap_context_menu_accelerator_key_t keys[4];
+} clap_context_menu_accelerator_t;
+
+// Context menu builder
+typedef struct clap_context_menu_builder {
+   void *ctx;
+
+   // accelerator may be null
+   bool(CLAP_ABI *add_entry)(const struct clap_context_menu_builder *builder,
+                             const char                             *label,
+                             const clap_context_menu_accelerator_t  *accelerator,
+                             bool                                    is_enabled,
+                             clap_id                                 action_id);
+
+   // accelerator may be null
+   bool(CLAP_ABI *add_checkbox)(const struct clap_context_menu_builder *builder,
```

Comment:
Shouldn't this be called `add_checkmark`, `add_checkmark_option` or `add_option` (since it makes a checkmark and not a checkbox)?

Reply:
Yeah maybe, I don't have a strong opinion on that.
To me checkbox is a toggle entry.
Options would be more if you have a set of entries which are exclusive to each other (radio buttons in some toolkits).

Reply:
Shouldn't that add a checkbox, and not a check mark?

A checkbox is one of these things: (https://en.wikipedia.org/wiki/Checkbox)

- [ ] Unchecked
- [x] Checked

This is a check mark: :heavy_check_mark: (https://en.wikipedia.org/wiki/Check_mark)

I'd assume this `add_checkbox()` option should add a checkbox that can be used to toggle a boolean value, right?

Reply:
So what builder option do you call when you don't have a check mark to display? The check mark option...but without the check mark? Like you said, other GUI usually call it just a 'check menu item'. So `add_check_entry()` would probably be the most consistent and least confusing option.